src/datasets/containers.jl

```diff
@@ -94,6 +308,21 @@ LearnBase.getobs(dataset::TableDataset{<:CSV.File}, idx) = dataset.table[idx]
 LearnBase.nobs(dataset::TableDataset{<:CSV.File}) = length(dataset.table)
 
 
+#TimeSeriesDataset
+
+struct TimeSeriesDataset{}
+    table::AbstractArray{Float64,3}
```

Comment:
```suggestion
struct TimeSeriesDataset{T<:AbstractArray}
    table::T
```
The type parameter is required for avoid an abstract field type. I've loosened the array type bound as well, but it may need to be tuned up or down later.